intermediate_source/parametrizations.py

```diff
@@ -0,0 +1,353 @@
+# Let's start by reimplementing the code above using ``torch.nn.utils.parametrize``.
+# The only thing that we have to do is to write the parametrization as a regular ``nn.Module``
+class Symmetric(nn.Module):
+    def forward(self, X):
+        return X.triu() + X.triu(1).transpose(-1, -2)
+
+###############################################################################
+# This is all we need to do. Once we have this, we can transform any regular layer into a
+# symmetric layer by doing
+layer = nn.Linear(3, 3)
+P.register_parametrization(layer, "weight", Symmetric())
+
+###############################################################################
+# Now, the matrix of the linear layer is symmetric
+A = layer.weight
+print(A)
+assert torch.allclose(A, A.T)
+
+###############################################################################
+# We can do the same thing with any other layer. For example, we can create a CNN with
+# skew-symmetric kernels (:math:`X = -X^{\intercal}`). We use a similar parametrization,
+# copying minus the upper triangular part into the lower-triangular part
+class Skew(nn.Module):
+    def forward(self, X):
+        A = X.triu(1)
+        return A - A.transpose(-1, -2)
+
+
+cnn = nn.Conv2D(in_channels=5, out_channels=8, kernel_size=3)
+P.register_parametrization(layer, "weight", Skew())
+# Print a few kernels
+print(cnn.weight[0, 1])
+print(cnn.weight[2, 2])
+
+###############################################################################
+# Inspecting a parametrized module
+# --------------------------------
+# When a module is parametrized, we find that the module has changed a bit.
+# We may observe this by simply printing the module
+layer = nn.Linear(3, 3)
+print(f"Unparametrized:\n{layer}")
+P.register_parametrization(layer, "weight", Symmetric())
+print(f"Parametrized:\n{layer}")
+
+###############################################################################
+# We see that the ``Symmetric`` class has been registered under a ``parametrizations`` attribute.
+# This ``parametrizations`` attribute is an ``nn.ModuleList``, and it can be accessed as such
+print(layer.parametrizations.weight[0])
+
+###############################################################################
+# Note that each element in the `ModuleList` is itself a list, and we have to select the first
+# element of this list. It will be clear later the reason for this, when we see how to contactenate
+# paramtrizations.
+#
+# Something that we may notice is that, if we print the parameters, we see that the
+# parameter ``weight`` has been moved
+print(dict(layer.named_parameters()))
+
+###############################################################################
+# It now sits under ``layer.parametrizations.weight.original``
+print(layer.parametrizations.weight.original)
+
+###############################################################################
+# Besides these two small differences, the parametrization is doing exactly the same
```

Comment:
I think it would be nice to specify which are these two from the beginning of the section:
"When a module is parametrized, we find that the module has changed a bit" -> When a module is parametrized, we find that the module has changed in two ways" or something similar?